chapters/feature-selection.qmd

```diff
@@ -0,0 +1,525 @@
+---
+knitr:
+  opts_chunk:
+    cache.path: "../_cache/feature-selection/"
+---
+
+# Feature Selection {#sec-feature-selection}
+
+```{r}
+#| label: feature-selection-setup
+#| include: false
+source("../R/_common.R")
+
+# ------------------------------------------------------------------------------
+
+library(heatmaply)
+library(future)
+library(tidymodels)
+library(colino)
+library(patchwork)
+
+# ------------------------------------------------------------------------------
+# Set options
+
+tidymodels_prefer()
+theme_set(theme_transparent())
+plan("multisession")
+set_options()
+
+# ------------------------------------------------------------------------------
+# load data and results
+
+source("../R/setup_deliveries.R")
+load("../RData/deliveries_cubist.RData")
+load("../RData/deliveries_lm.RData")
+load("../RData/forested_data.RData")
+
+ctrl_rs <- control_resamples(
+  save_pred = TRUE,
+  parallel_over = "everything",
+  save_workflow = TRUE
+)
+
+reg_metrics <- metric_set(mae, rmse, rsq)
+```
+
+In Chapters [-@sec-numeric-predictors] through [-@sec-interactions-nonlinear], we described different methods for adding new predictors to the model (e.g., interactions, spline columns, etc.). We can add any predictors (or functions of predictors) based on first principles or our intuition or contextual knowledge. 
+
+We are also motivated to remove unwanted or irrelevant predictors. This can decrease model complexity, training time, computer memory, the cost of acquiring unimportant predictors, and so on. For some models, the presence of non-informative predictors can _decrease_ performance. 
+
+We may want to remove predictors based on their distributions. Unsupervised feature selection focuses on just the predictor data. 
+
+More often, though, we are interested in supervised feature selection^[Recall that we use the terms "predictors" and "features" interchangeably. We favor the former term, but "features" is more commonly used when discussing the filtering of columns.]: the removal of predictors that do not appear to have a relationship to the outcome. This implies that we have a statistic that can compute some measure of each predictor’s importance. 
```

Comment:
> This implies that we have a statistic that can compute some measure of each predictor’s importance.

Does it tho? I feel like we don't **technically** have a importance measure in wrapper methods. At least not in the traditional sense.

we have a measure for each subset of predictors, but i don't think we have a natural meassure one for each predictor itself.

this feels very nit-picky